src/tests/ftest/launch.py

```diff
@@ -1342,8 +1364,23 @@ def install_debuginfos():
 
     cmds.append(cmd)
 
+    retry = False
     for cmd in cmds:
-        print(get_output(cmd))
+        try:
+            print(run_command(cmd))
+        except RuntimeError as error:
+            # got an error, so abort this list of commands and re-run
+            # it with a yum clean, makecache first
+            print(error)
+            retry = True
+            break
+    if retry:
+        print("Going to refresh caches and try again")
+        cmd_prefix = ["sudo", "yum", "--enablerepo=*debug*"]
+        cmds.insert(0, cmd_prefix + ["clean", "all"])
+        cmds.insert(1, cmd_prefix + ["makecache"])
+        for cmd in cmds:
+            print(run_command(cmd))
```

Comment:
Shouldn't there be a timeout or max number of retries? In a situation where e.g. the yum servers are wonky or there's some other network problem, couldn't this result in a test hanging forever (or at least until the problem resolves, potentially hours later)?

Reply:
I don't think there is any looping here, unless I am missing something.  The logic as I intend it, is to try the debuginfo-install commands once.  If there is an error with that iteration, prefix the commands with a _clean_ and run the commands a second time, only.  Without even writing a loop around it to do it.

Reply:
Even though this has landed @mjmac I'm still interested in knowing if you see a hanging/looping situation that I have not seen in the PR so that we can address it if so.

Reply:
Hmm, no, you're right. I misread the loop over the commands as an outer loop around the logic. And thinking about it a bit more, even if there were an outer loop, since this is launch.py it should be constrained by the overall test timeout. No concerns, thanks for responding.